File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/configs/CorsConfig.java
package com.chaincuerealestate.chaincuerealestate.configs;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.Ordered;
import org.springframework.core.annotation.Order;
import org.springframework.http.CacheControl;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
@ConfigurationProperties(prefix = "teacher-portal-client.cors")
@Getter
@Setter
@Order(Ordered.HIGHEST_PRECEDENCE)
public class CorsConfig implements WebMvcConfigurer {

    private static final String[] ALLOWED_HEADERS = {"Authorization", "content-type"};
    private static final String[] ALLOWED_METHODS = {"GET", "PUT", "POST", "DELETE", "OPTIONS"};
    private String allowedOrigin = "*";
    private int maxAge = 3600;

    @Override
    public void addCorsMappings(CorsRegistry registry) {
        registry.addMapping("/**")
                .allowedOrigins(allowedOrigin.split(","))
                .allowedMethods(ALLOWED_METHODS)
                .allowedHeaders(ALLOWED_HEADERS)
                .maxAge(maxAge);
    }

    @Override
    public void addResourceHandlers(ResourceHandlerRegistry registry) {
        registry.addResourceHandler("*/**")
                .setCacheControl(CacheControl.noCache());
    }
}

File: kotlin/chaincue-real-estate/src/main/kotlin/com/example/chaincuerealestate/configs/CorsConfig.kt
package com.example.chaincuerealestate.configs

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Configuration
import org.springframework.core.Ordered
import org.springframework.core.annotation.Order
import org.springframework.http.CacheControl
import org.springframework.web.servlet.config.annotation.CorsRegistry
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer


@Configuration
@ConfigurationProperties(prefix = "chaincue-real-estate.cors")
@Order(Ordered.HIGHEST_PRECEDENCE)
class CorsConfig : WebMvcConfigurer {

    companion object {
        val ALLOWED_HEADERS: Array<String> = arrayOf("Authorization", "content-type")
        val ALLOWED_METHODS: Array<String> = arrayOf("GET", "PUT", "POST", "DELETE", "OPTIONS")
    }

    var allowedOrigin = "*"
    var maxAge = 3600

    override fun addCorsMappings(registry: CorsRegistry) {
        registry.addMapping("/**")
            .allowedOrigins(*allowedOrigin.split(",").toTypedArray())
            .allowedMethods(*ALLOWED_METHODS)
            .allowedHeaders(*ALLOWED_HEADERS)
            .maxAge(maxAge.toLong())
    }

    override fun addResourceHandlers(registry: ResourceHandlerRegistry) {
        registry.addResourceHandler("*/**").setCacheControl(CacheControl.noCache())
    }
}
